safehttp/incoming_request.go

```diff
@@ -29,15 +28,15 @@ type IncomingRequest struct {
 	Header    Header
 	parseOnce sync.Once
 	TLS       *tls.ConnectionState
-	URL       *url.URL
+	URL       URL
```

Comment:
Why are we not using a pointer here? This will be a choice we’re stuck with.

Reply:
I changed it to be a pointer. Why is it better if it's a pointer?

Reply:
Because then we have the option to declare methods with a pointer receiver later on, allowing us to modify fields of our URL struct, if we ever need to.

If we don’t use a pointer here, we don’t have that option.